### Monfero John Benedict
#### Formative Assessment 6

# Q1: Determine the Quartiles in Grouped Dataset

### Given Data:
- Grade intervals and the number of students:

| Grade Interval | Number of Students |
|----------------|--------------------|
| 90-100         | 9                  |
| 80-89          | 32                 |
| 70-79          | 43                 |
| 60-69          | 21                 |
| 50-59          | 11                 |
| 40-49          | 3                  |
| 30-39          | 1                  |

In [6]:
import numpy as np

# Raw data (grade intervals and number of students)
grade_intervals = [(90, 99), (80, 89), (70, 79), (60, 69), (50, 59), (40, 49), (30, 39)]
num_students = [9, 32, 43, 21, 11, 3, 1]

# Create a list to store the grade values (assuming midpoint of each interval as representative value)
grades = []

for (low, high), count in zip(grade_intervals, num_students):
    midpoint = (low + high) / 2
    grades.extend([midpoint] * count)

# Sort the grades for quartile calculation
grades = np.array(sorted(grades))

# Calculate Q1, Q2 (median), and Q3
Q1 = np.percentile(grades, 25)
Q2 = np.percentile(grades, 50)
Q3 = np.percentile(grades, 75)

(Q1, Q2, Q3)

(64.5, 74.5, 84.5)

### Total number of students: 120

#### Quartiles:

- **Q1 (25th percentile)**: 64.5
- **Q2 (50th percentile / Median)**: 74.5
- **Q3 (75th percentile)**: 84.5

### Data Intepretation

| Grade Interval | Number of Students |
|----------------|--------------------|
| 90-99          | 9                  |
| 80-89          | 32                 |
| 70-79          | 43                 |
| 60-69          | 21                 |
| 50-59          | 11                 |
| 40-49          | 3                  |
| 30-39          | 1                  |

**The Quartile Distribution Determines the following**

1. The $ Q_1 $ class located on the grade interval 60 - 69
2. The $ Q_2 $ class located on the grade interval 70 - 79
3. The $ Q_3 $ class located on the grade interval 80 - 89

# Q2: Dispersion Comparison between Two Subjects

#### On a final examination in statistics, the mean grade of a group of 150 students was 78 and the standard deviation was 8.0. In algebra, however, the mean final grade of the group was 73 and the standard deviation was 7.6. In which subject was there the greater (a) absolute dispersion and (b) relative dispersion?

The second question asks to compare **absolute and relative dispersion** of grades in two subjects based on their means and standard deviations:

    - We know that absolute dispersion refers to the standard deviation (SD). We directly compare the standard deviations.
    
$$ SD = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \overline{x})^2} $$

    - Relative dispersion, on the other hand, is typically measured using the coefficient of variation (CV), defined as:

$$ CV = \frac{\sigma}{μ} \cdot 100 $$

In [16]:
# Given data
mean_stats = 78
std_dev_stats = 8
mean_algebra = 73
std_dev_algebra = 7.6

# Coefficient of variation calculation
cv_stats = (std_dev_stats / mean_stats) * 100
cv_algebra = (std_dev_algebra / mean_algebra) * 100

# Output results for absolute and relative dispersion
print(f"""
Data Result:
    Standard Deviation among students' final grade in statistics:
        {std_dev_stats: .3f}
    Standard Deviation among students' final grade in algebra:
        {std_dev_algebra: .3f}
    Coefficient of Variation among students' final grade in statistics:
        {cv_stats: .3f}
    Coefficient of Variation among students' final grade in algebra:
        {cv_algebra: .3f}
    """)


Data Result:
    Standard Deviation among students' final grade in statistics:
         8.000
    Standard Deviation among students' final grade in algebra:
         7.600
    Coefficient of Variation among students' final grade in statistics:
         10.256
    Coefficient of Variation among students' final grade in algebra:
         10.411
    


In [23]:
# Compare which are bigger dispersions in terms of SD and CV:
if std_dev_stats > std_dev_algebra and cv_stats < cv_algebra:
    print(f"""
The final grades among students is more absolute disperse/spread (SD) in statistics as compared to algebra by {std_dev_stats - std_dev_algebra : .3f}

However, the relative dispersion, the final grades among student's algebra has the greater spread by {cv_algebra - cv_stats : .3f}
""")


The final grades among students is more absolute disperse/spread (SD) in statistics as compared to algebra by  0.400

However, the relative dispersion, the final grades among student's algebra has the greater spread by  0.155



### Data Interpretation

**The final grades among students is more absolute disperse/spread (SD) in statistics as compared to algebra by  0.400**

**However, the relative dispersion, the final grades among student's algebra has the greater spread by  0.155**

# Q3: Evaluate the Standard Scores

#### Prove that the mean and standard deviation of a set of standard scores are equal to 0 and 1, respectively. Use the following problem to illustrate this: Convert the set 6, 2, 8, 7, 5 into standard scores.

### Z-score formula

$$ z = \frac{x - μ}{\sigma}$$

In [24]:
# Raw data for question 3
data = np.array([6, 2, 8, 7, 5])

# Calculate mean and standard deviation of raw data
mean_data = np.mean(data)
std_dev_data = np.std(data)

# Convert to z-scores
z_scores = (data - mean_data) / std_dev_data

# Calculate mean and standard deviation of z-scores
mean_z_scores = np.mean(z_scores)
std_dev_z_scores = np.std(z_scores)

# Output z-score mean and standard deviation
print(f"""
Data Result:
The mean of the z-scores: {mean_z_scores : .3f}
The standard deviation of the z-scores: {std_dev_z_scores : .3f}
""")


Data Result:
The mean of the z-scores:  0.000
The standard deviation of the z-scores:  1.000



### Data Interpretation

**The given raw datapoints proves that the mean of their z-scores is always 0 that is:**

$$ μ_{z-scores} = 0 $$

**Moreover, the standard deviation of their z-scores is always 1 that is:**

$$ \sigma_{z-scores} = 1 $$

# Q4: Mean and Standard Deviation of the Sum of Masses

#### Three masses are measured as 20.48, 35.97, and 62.34 g, with standard deviations of 0.21, 0.46, and 0.54 g, respectively. Find the (a) mean and (b) standard deviation of the sum of the masses.

### The mean of the sum of the means
$$ \mu_{\text{sum}} = \mu_1 + \mu_2 + \mu_3 $$
### Standard Deviation of the sum of the standard deviations
$$ \sigma_{\text{sum}} = \sqrt{\sigma_1^2 + \sigma_2^2 + \sigma_3^2} $$

In [11]:
# Given data for masses and their standard deviations
masses = [20.48, 35.97, 62.34]
std_devs = [0.21, 0.46, 0.54]

# Calculate the mean of the sum of the masses
mean_sum_masses = sum(masses)

# Calculate the standard deviation of the sum of the masses
variance_sum_masses = sum([std_dev**2 for std_dev in std_devs])
std_dev_sum_masses = np.sqrt(variance_sum_masses)

(mean_sum_masses, std_dev_sum_masses)

(118.79, 0.7397972695272672)

### Data Interpretation

#### The mean of the sum of the means:
$$ \mu_{\text{sum}} = 118.79 $$

#### The standard deviation of the sum of the standard deviations:
$$ \sigma_{\text{sum}} = 0.740 $$

# Q5: Credit Hour Distribution at Metropolitan Technological College

### Given Data:

| Credit Hours (x) | Probability p(x) |
|------------------|------------------|
| 6                | 0.1              |
| 9                | 0.2              |
| 12               | 0.4              |
| 15               | 0.2              |
| 18               | 0.1              |

### Task:
1. Find the **mean (μ)** and **variance (σ²)**.
2. List the 25 possible samples of size 2, their means, and their probabilities.


In [26]:
import numpy as np
import itertools
import pandas as pd

# Given data
credit_hours = np.array([6, 9, 12, 15, 18])
probabilities = np.array([0.1, 0.2, 0.4, 0.2, 0.1])

# Part 1: Calculate the mean (μ) and variance (σ²)
mean = np.sum(credit_hours * probabilities)  # μ = Σ(x * p(x))
variance = np.sum(((credit_hours - mean) ** 2) * probabilities)  # σ² = Σ((x - μ)² * p(x))

# Part 2: List the 25 possible samples of size 2, their means, and probabilities
samples = list(itertools.product(credit_hours, repeat=2))
sample_means = [(x[0] + x[1]) / 2 for x in samples]
sample_probs = [probabilities[credit_hours == x[0]][0] * probabilities[credit_hours == x[1]][0] for x in samples]

# Create a DataFrame
results_df = pd.DataFrame({
    'Sample': samples,
    'Mean': sample_means,
    'Probability': sample_probs
})

# Display the DataFrame
results_df


Sample  Mean  Probability
0     (6, 6)   6.0         0.01
1     (6, 9)   7.5         0.02
2    (6, 12)   9.0         0.04
3    (6, 15)  10.5         0.02
4    (6, 18)  12.0         0.01
5     (9, 6)   7.5         0.02
6     (9, 9)   9.0         0.04
7    (9, 12)  10.5         0.08
8    (9, 15)  12.0         0.04
9    (9, 18)  13.5         0.02
10   (12, 6)   9.0         0.04
11   (12, 9)  10.5         0.08
12  (12, 12)  12.0         0.16
13  (12, 15)  13.5         0.08
14  (12, 18)  15.0         0.04
15   (15, 6)  10.5         0.02
16   (15, 9)  12.0         0.04
17  (15, 12)  13.5         0.08
18  (15, 15)  15.0         0.04
19  (15, 18)  16.5         0.02
20   (18, 6)  12.0         0.01
21   (18, 9)  13.5         0.02
22  (18, 12)  15.0         0.04
23  (18, 15)  16.5         0.02
24  (18, 18)  18.0         0.01

### Data Result and Interpretation (A)

#### The mean of the sample is 12.0 and the variance of the sample is 10.8

In [29]:
# Gather Results
summary_df = pd.DataFrame({
    'Mean': [mean],
    'Variance': [variance]
})

summary_df = summary_df.rename_axis('Measure of Statistics')
summary_df

Mean  Variance
Measure of Statistics                
0                      12.0      10.8

### Data Result and Interpretation (B)
#### The probability distribution among 25 sample combinations, fluctuates from 0.1 to 0.16

In [31]:
# Gather Results
results_df = pd.DataFrame({
    'Sample': samples,
    'Mean': sample_means,
    'Probability': sample_probs
})

results_df = results_df.rename_axis('Possible Sample ID')
results_df

Sample  Mean  Probability
Possible Sample ID                             
0                     (6, 6)   6.0         0.01
1                     (6, 9)   7.5         0.02
2                    (6, 12)   9.0         0.04
3                    (6, 15)  10.5         0.02
4                    (6, 18)  12.0         0.01
5                     (9, 6)   7.5         0.02
6                     (9, 9)   9.0         0.04
7                    (9, 12)  10.5         0.08
8                    (9, 15)  12.0         0.04
9                    (9, 18)  13.5         0.02
10                   (12, 6)   9.0         0.04
11                   (12, 9)  10.5         0.08
12                  (12, 12)  12.0         0.16
13                  (12, 15)  13.5         0.08
14                  (12, 18)  15.0         0.04
15                   (15, 6)  10.5         0.02
16                   (15, 9)  12.0         0.04
17                  (15, 12)  13.5         0.08
18                  (15, 15)  15.0         0.04
19                  (15, 18)  16.5         0.02
20                   (18, 6)  12.0         0.01
21                   (18, 9)  13.5         0.02
22                  (18, 12)  15.0         0.04
23                  (18, 15)  16.5         0.02
24                  (18, 18)  18.0         0.01

In [45]:
# Group by mean and sum the probabilities
grouped_df = results_df.groupby('Probability', as_index=False).agg({'Sample': 'count'})
grouped_df = grouped_df.rename_axis('Probability Result ID')
grouped_df

Probability  Sample
Probability Result ID                     
0                             0.01       4
1                             0.02       8
2                             0.04       8
3                             0.08       4
4                             0.16       1

### Data Interpretation (B)

#### 8 out of 25 samples has either the probability outcome of 0.02 or 0.04

#### 4 out of 25 samples has either the probability outcome of 0.01 or 0.08

#### The sample (12,12) delegates a probability outcome of 0.16